ui/src/wishDetails/WishDetails.js

```diff
@@ -1,30 +1,102 @@
 import React, { Component } from 'react';
+import WishDetailsService from '../services/WishDetailsService'
+import Rocket from '../assets/icn_To_Go_Rocket_White_Inside_130x130.png'
+import Alien from '../assets/icn_To_Meet_Alien_White_Inside_130x130.png'
+import Astronaut from '../assets/icn_To_Be_Astronaut_White_Inside_130x130.png'
+import Telescope from '../assets/icn_To_See_Telescope_White_Inside_130x130.png'
 
 export default class WishDetails extends Component {
   constructor(props) {
     super();
     this.state = {
-      id: 0
+      id: '',
+      wishDetails: {
+        id: '',
+        child: {
+          firstName: '',
+          lastName: ' ',
+          hometown: '',
+          illness: '',
+          age: ''
+        },
+        type: '',
+        details: '',
+        sponsor: {
+          name: '',
+          logo: '',
+          links: []
+        },
+        createdAt: '',
+        updatedAt: ''
+      }
     }
   }
 
-  componentDidMount() {
-
+  async componentDidMount() {
+    const wishDetails = await WishDetailsService.getWishDetails(this.state.id)
+    this.setState({
+      wishDetails: wishDetails
+    })
   }
 
+  getImageByType = () => {
+    let type = this.state.wishDetails.type.toUpperCase();
+    let image = '';
+    switch (type) {
+      case 'GO':
+        image = Rocket;
+        break;
+      case 'MEET':
+        image = Alien;
+        break;
+      case 'BE':
+        image = Astronaut;
+        break;
+      case 'SEE':
+        image = Telescope;
+        break;
+      default:
+        break;
+    }
+    return image;
+  }
 
   render() {
+    const { child, details, sponsor } = { ...this.state.wishDetails }
```

Comment:
Not sure the additional object destructuring on the right hand side is necessary.

This should be sufficient
```javascript
const { child, details, sponsor } = this.state.wishDetails
```

Reply:
Yeah you’re right, I changed it while I was debugging to see something and forgot to change it back

Reply:
Fixed this in latest commit